### DISCUSSION TITLE
WebRTC Url

### DISCUSSION THREAD

**Discussion by pelaxa** [koush/scrypted#1145] [Ideas]
Hi,
I cannot figure out what the WebRTC URL is for my cameras.  I have it all configured and working, and I can view the RTSP streams, but not sure what to set in MagicMirror WebRTC client to see a specific camera.  what am I missing here?

**Comment by bjia56** [comment]
Scrypted does not publish a public webrtc url - it's not really possible to do in a portable way since the signaling channel required by webrtc is application-specific. You may want to see if MagicMirror supports consuming an RTSP url, which is available with the scrypted rebroadcast plugin

**Reply by pelaxa** [reply]
Thanks.  so the WebRTC stream is not consumable outside of scrypted itself?  I thought that was the point of the plugin and that is how it feeds HomeAssistant as well!  

The plugin says: "This plugin sends video feeds from Scrypted cameras to WebRTC clients such as Scrypted’s management console, Chromecast, Google Home, Alexa, and others."

I guess I am not clear how these clients can use the stream and how I could do the same for MagicMirror.  I am running MagicMirror client on a Google Chromecast device so I should be able to leverage it.

**Reply by bjia56** [reply]
Each plugin has its own integration with Scrypted's webrtc, so the plugins understand how to translate signaling channels from external services into Scrypted's signaling channel. MagicMirror would need its own Scrypted plugin to use its webrtc.

**Reply by bjia56** [reply]
Webrtc itself is a web standard which defines how two peers can send media traffic from one to the other, but for the peers to negotiate that connection, a lightweight signaling channel has to be used to send text traffic. The standard does not dictate the kind of signaling channel that has to be used, so it can be different for each application, which is why Scrypted webrtc is only usable if something can speak the same Scrypted API language as the plugins.

I think you'd be better off looking at integration with the RTSP stream, and see if there's software that can convert that into webrtc usable by MagicMirror.

**Reply by pelaxa** [reply]
Thanks for the explanation.  

Unfortunately, RTSP does not seem to work no matter what I tried on MagicMirror.  Before using scrypted, I was using go2RTC which allowed me to show the video on MagicMirror using WebRTC, but I was hoping to consolidate and not add more to the network.

Based on what you said, then it should be possible to look at one of these clients and possibly create a plugin for MM to leverage Scrypted.  a bigger task than I was hoping it would take, but seems like the only route.

**Reply by pelaxa** [reply]
One last question on this:  is it not possible for Scrypted to provide an HTML page that would load its webRTC client and show the steam for a particular camera?  then third parties could just load that HTML page to show the stream?  

I can make a separate feature request if this is a possibility.

**Reply by bjia56** [reply]
Yes, that's theoretically possible to do. I'm not sure if there's anything existing in Scrypted that has that functionality today though.

**Reply by piotrtobolski** [reply]
There is functionality like this but it is part of paid Scrypted NVR and it is used to create Home Assistant card. It should be possible to use that outside HA as well as is is webpage in an iframe. Docs: https://docs.scrypted.app/home-assistant.html

Scrypted author isn't willing to create lower paid tier for people who want just the card and don't need the whole NVR. (I was asking him in one of the channels on Discord).
